src/NuGet.Core/NuGet.Packaging/PackageCreation/Xml/PackageMetadataXmlExtensions.cs

```diff
@@ -374,5 +374,17 @@ private static void AddElementIfNotNull<T>(XElement parent, XNamespace ns, strin
                 }
             }
         }
+
+        private static void AddElementIfNotEmpty<T>(XElement parent, XNamespace ns, string name, IEnumerable<T> value, Func<IEnumerable<T>, object> process)
```

Comment:
I don't see much difference between `AddElementIfNotEmpty` and `AddElementIfNotNull` method implementation. Can we rename the existing method from `AddElementIfNotNull` to `AddElementIfNotNullAndEmpty`?.
```cs
private static void AddElementIfNotNullAndEmpty<T>(XElement parent, XNamespace ns, string name, T value, Func<T, object> process)
            where T : class
        {
            if (value != null && value.Any())
            {
                var processed = process(value);
                if (processed != null)
                {
                    parent.Add(new XElement(ns + name, processed));
                }
            }
        }
```

Reply:
@kartheekp-ms I don't think that will work. There is no `.Any()` method defined for any class, only for types that extend `IEnumerable<T>`. Unless I misunderstand something, I think your proposal will fail to compile.

Reply:
Thanks for the update @zivkan. I overlooked the fact that `value` is `IEnumerable<T>` in `AddElementIfNotEmpty` method.

Reply:
`(value as IEnumerable)?.Cast<object>().Any() != false` could work, or a more expanded version that calls MoveNext directly.

Reply:
I think it will be easier to understand if left as two separate methods, as you already have. Reducing lines of code by making it harder to understand isn't a good outcome.